org-code-javabuilder/lib/src/main/java/org/code/javabuilder/UserProjectFileManager.java

```diff
@@ -1,15 +1,65 @@
 package org.code.javabuilder;
 
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.*;
+
+/** Manages the in-memory list of user project files. */
 public class UserProjectFileManager implements ProjectFileManager {
+  private final String baseUrl;
+  private final String[] fileNames;
+  private final ArrayList<ProjectFile> fileList;
 
-  public UserProjectFileManager(String baseUrl, String[] fileNames) {}
+  public UserProjectFileManager(String baseUrl, String[] fileNames) {
+    this.baseUrl = baseUrl;
+    this.fileNames = fileNames;
+    this.fileList = new ArrayList<>();
+  }
 
+  /**
+   * Asynchronously loads all user project files.
+   *
+   * @throws UserFacingException If the load times out or fails.
+   */
   public void loadFiles() throws UserFacingException {
-    throw new UserFacingException("We couldn't load your files");
+    ExecutorService executor = Executors.newCachedThreadPool();
```

Comment:
Did you want to bound the number of concurrent requests to dashboard?  If so, I think using newFixedThreadPool() might be more appropriate.

Reply:
Hmm what's the concern you're addressing here?

Reply:
If you have lots of files, that would result in a potentially unbounded number of parallel requests to dashboard.

Reply:
Looks like this may end up being a moot point. We just started looking into how we'd handle multiple files, and it's looking like we'll store all files for an individual user project in a single json blob. This will end up being a single request (or possibly two depending on our final design).